### PR TITLE
clean up waitgroup and attach logic

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@soltysh @mfojtik @gabemontero @csrwng ptal.  i'm still testing this locally but i think it's the source of our hangs.

two issues it addresses:
1) waitgroup objects need to be passed by ref.
2) i didn't see any reason for containerAttachTwo?  containerAttachOne was already attaching stderr so it seemed redundant.  but i'd love a second opinion on that.

**Comment by bparees** [comment]
apparently that second attach gets us the output of the assemble script..  putting it back in and testing if things still hang.  hopefully the real issue here is the wg pointer, which would make sense since it would coincide with the changes to add the "run after build" feature which is when i started noticing these hangs.

**Comment by csrwng** [comment]
The reason we do it in 2 calls was because there was a bug in Docker attaching stdin, stdout and stderr all at once... you may want to remove the if/else in containerAttachOne to see if we can do without the 2 methods. Also call it something other than containerAttachOne if you're going to remove the other one :-)

**Comment by bparees** [comment]
@csrwng yeah i saw the bug comment, but it still looked like attachone was attaching them all at once anyway.  (and yet based on the observed behavior when i removed it, that's apparently not the case)

**Comment by bparees** [comment]
ok this seems to be sufficient to fix the hangs i was getting.

**Comment by csrwng** [comment]
@bparees it's either or:
https://github.com/openshift/source-to-image/blob/master/pkg/docker/docker.go#L422-L428

**Comment by csrwng** [comment]
LGTM

**Comment by gabemontero** [comment]
lgtm

**Comment by bparees** [comment]
[merge]

File: pkg/docker/docker.go [hunk 1]
```diff
@@ -528,22 +506,11 @@ func (d *stiDocker) RunContainer(opts RunContainerOptions) (err error) {
 	defer d.RemoveContainer(container.ID)
 
 	glog.V(2).Infof("Attaching to container")
-	// creating / piping the channels in runContainerAttachOne lead to unintended hangs
+	// creating / piping the channels in runContainerAttach lead to unintended hangs
 	attached := make(chan struct{})
-	wg := runContainerAttachOne(attached, container, opts, d)
+	wg := runContainerAttach(attached, container, opts, d)
 	attached <- <-attached
```
Comment (liggitt) [inline pkg/docker/docker.go]: why are we feeding the output of the attached channel back into attached channel?

Comment (liggitt) [inline pkg/docker/docker.go]: nevermind, I read the docs

**Comment by soltysh** [comment]
Nice job @bparees :+1:

**Pull request merged**